dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AbstractWorkflowResource.java

```diff
@@ -552,30 +558,35 @@ private Workflow addDockstoreYmlVersionToWorkflow(String repository, String gitR
-                existingWorkflowVersion.get().setAliases(remoteWorkflowVersion.getAliases());
-                existingWorkflowVersion.get().setSubClass(remoteWorkflowVersion.getSubClass());
-                existingWorkflowVersion.get().setCommitID(remoteWorkflowVersion.getCommitID());
-                existingWorkflowVersion.get().setDagJson(null);
-                existingWorkflowVersion.get().setToolTableJson(null);
-                existingWorkflowVersion.get().setReferenceType(remoteWorkflowVersion.getReferenceType());
-                existingWorkflowVersion.get().setValid(remoteWorkflowVersion.isValid());
-
-                updateDBVersionSourceFilesWithRemoteVersionSourceFiles(existingWorkflowVersion.get(), remoteWorkflowVersion);
+                existingWorkflowVersion.setWorkflowPath(remoteWorkflowVersion.getWorkflowPath());
+                existingWorkflowVersion.setLastModified(remoteWorkflowVersion.getLastModified());
+                existingWorkflowVersion.setLegacyVersion(remoteWorkflowVersion.isLegacyVersion());
+                existingWorkflowVersion.setAliases(remoteWorkflowVersion.getAliases());
+                existingWorkflowVersion.setSubClass(remoteWorkflowVersion.getSubClass());
+                existingWorkflowVersion.setCommitID(remoteWorkflowVersion.getCommitID());
+                existingWorkflowVersion.setDagJson(null);
+                existingWorkflowVersion.setToolTableJson(null);
+                existingWorkflowVersion.setReferenceType(remoteWorkflowVersion.getReferenceType());
+                existingWorkflowVersion.setValid(remoteWorkflowVersion.isValid());
+                updateDBVersionSourceFilesWithRemoteVersionSourceFiles(existingWorkflowVersion, remoteWorkflowVersion);
             } else {
                 workflow.addWorkflowVersion(remoteWorkflowVersion);
             }
 
-            Optional<WorkflowVersion> addedVersion = workflow.getWorkflowVersions().stream().filter(workflowVersion -> Objects.equals(workflowVersion.getName(), remoteWorkflowVersion.getName())).findFirst();
-            addedVersion.ifPresent(workflowVersion -> gitHubSourceCodeRepo
-                    .updateVersionMetadata(workflowVersion.getWorkflowPath(), workflowVersion, workflow.getDescriptorType(), repository));
+            WorkflowVersion addedVersion = workflowVersionDAO.getWorkflowVersionByWorkflowIdAndVersionName(workflow.getId(), remoteWorkflowVersion.getName());
+            if (addedVersion != null) {
+                gitHubSourceCodeRepo.updateVersionMetadata(addedVersion.getWorkflowPath(), addedVersion, workflow.getDescriptorType(), repository);
+
+                // Update file formats for the version and then the entry.
+                // TODO: We were not adding file formats to .dockstore.yml versions before, so this only handles new/updated versions. Need to add a way to update all .dockstore.yml versions in a workflow
+                Set<WorkflowVersion> workflowVersions = new HashSet<>();
```

Comment:
Your code is more than fine and I wouldn't change it, but FYI, I just discovered you can do `Set.of(addedVersion)` as of Java 9.